content/en/blog/_posts/2021-03-30-Kubernetes-Multi-Tenancy-Usage-Models.md

```diff
@@ -0,0 +1,80 @@
+
+To isolate tenant workloads, each namespace must also contain:
+* **[role bindings](/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding):** for controlling access to the namespace
+* **[network policies](/docs/concepts/services-networking/network-policies/):** to prevent network traffic across tenants
+* **[resource quotas](/docs/concepts/policy/resource-quotas/):** to limit usage and ensure fairness across tenants
+
+With this model, tenants share cluster-wide resources like ClusterRoles and CustomResourceDefinitions (CRDs) and hence cannot create or update these cluster-wide resources.
+
+The [Hierarchical Namespace Controller (HNC)](https://kubernetes.io/blog/2020/08/14/introducing-hierarchical-namespaces/) project makes it easier to manage namespace based multi-tenancy by allowing users to create additional namespaces under a namespace, and propagating resources within the namespace hierarchy. This allows self-service namespaces for tenants, without requiring cluster-wide permissions.
+
+The [Multi-Tenancy Benchmarks (MTB)](https://github.com/kubernetes-sigs/multi-tenancy/tree/master/benchmarks) project provides benchmarks and a command-line tool that performs several configuration and runtime checks to report if tenant namespaces are properly isolated and the necessary security controls are implemented.
+
+### Clusters-as-a-Service
+
+With the *clusters-as-a-service* usage model, each tenant gets their own cluster.  This model allows tenants to have different versions of cluster-wide resources, such as CRDs.
+
+The tenant clusters may be provisioned using projects like [Cluster API (CAPI)](https://cluster-api.sigs.k8s.io/) where a management cluster is used to provision multiple workload clusters. A workload cluster is assigned to a tenant and tenants have full control over cluster resources. Note that in most enterprises a central platform team may be responsible for managing required add-on services such as security and monitoring services, and for providing cluster lifecycle management services such as patching and upgrades. A tenant administrator may be restricted from modifying the centrally managed services and other critical cluster information. 
+
+In a variation of the *clusters-as-a-service* model, the tenant cluster may be a **virtual cluster**. As with other forms of virtualization, users of a virtual cluster see no significant differences between a virtual cluster and other Kubernetes clusters. However, a virtual cluster shares worker node resources and some workload state independent control plane components, like the scheduler. Other workload aware control-plane components, like the API server, are created on a per-tenant basis and additional components are used to synchronize and manage state across virtual clusters and the underlying shared cluster. This provides each tenant their own virtual control-plane that can operate independently of other tenants.
```

Comment:
Does it make sense to pull this into its own section, e.g. "virtual-cluster per team"?

Reply:
I had started with 3 options, but then felt that virtual clusters was just a implementation detail of "clusters-as-a-service" for the consumer, and so combined the two. Open to splitting again, if there is agreement that these should be separate.

Reply:
IMO VCs have enough differences with truly distinct clusters that they deserve their own section. Most notably, separate clusters cannot both place pods on the same node at the same time, while VCs can, so you need to take much more care to protect the data plane. But I'll go with the majority opinion on this one.